addons/box/models.py

```diff
@@ -99,10 +99,7 @@ def revoke_remote_oauth_access(self, external_account):
 
     @staticmethod
     def sync_with_gravyvalet(data_from_gv, owner, is_deleted):
-        try:
-            settings_obj = UserSettings.objects.get(owner=owner)
-        except ObjectDoesNotExist:
-            return None  # or handle the case where the settings object does not exist
+        settings_obj = UserSettings.objects.get_or_create(owner=owner)
```

Comment:
drive by: this belongs in NodeSettings, the UserSettings rarely contain anything consequential for us (pretty much just a super janky dict mapping nodes to external accounts that we've replaced with actual database relationships on the gv side)